cmd/provision.go

```diff
@@ -60,26 +66,35 @@ This ensures that your project ports are properly exposed and externally accessi
 			log.Fatal("remote with name already exists")
 		}
 
-		// Load flags
+		// Load flags for credentials
 		fromEnv, _ := cmd.Flags().GetBool("from-env")
-		instanceType, _ := cmd.Flags().GetString("type")
+		profilePath, _ := cmd.Flags().GetString("profile-path")
+
+		// Load flags for setup configuration
 		user, _ := cmd.Flags().GetString("user")
+		instanceType, _ := cmd.Flags().GetString("type")
 		stringProjectPorts, _ := cmd.Flags().GetStringArray("ports")
 
 		// Create VPS instance
 		var prov *provision.EC2Provisioner
-		if !fromEnv {
-			var id, key string
-			id, key, err = enterEC2CredentialsWalkthrough(os.Stdin)
+		if fromEnv {
+			prov, err = provision.NewEC2ProvisionerFromEnv(os.Stdout)
 			if err != nil {
 				log.Fatal(err)
 			}
-			prov, err = provision.NewEC2Provisioner(id, key, os.Stdout)
+		} else if profilePath != "" {
+			profileUser, _ := cmd.Flags().GetString("profile-user")
+			prov, err = provision.NewEC2ProvisionerFromProfile(profilePath, profileUser, os.Stdout)
 			if err != nil {
 				log.Fatal(err)
 			}
 		} else {
-			prov, err = provision.NewEC2ProvisionerFromEnv(os.Stdout)
+			var id, key string
+			id, key, err = enterEC2CredentialsWalkthrough(os.Stdin)
```

Comment:
Is there a particular reason why you separated variable declaration? I thought you can do things like `id, key, err := ...`

Reply:
Good point, I think this was a holdover from something else I had intended to do, where `id, key` could be set from different functions depending on the condition. I'll simplify it